src/compas_fab/backends/kinematics/analytical_inverse_kinematics.py

```diff
@@ -0,0 +1,143 @@
+
+        solutions = self._inverse_kinematics(frame_RCF)
+
+        configurations = self.joint_angles_to_configuration(robot, solutions)
+
+        # check collisions for all configurations (sets those to `None` that are not working)
+        if options and "check_collision" in options and options["check_collision"] is True:
+            for i, config in enumerate(configurations):
+                try:
+                    self.client.check_collisions(robot, config)
+                except BackendError:
+                    configurations[i] = None
+
+        # fit configurations within joint bounds (sets those to `None` that are not working)
+        configurations = self.try_to_fit_configurations_between_bounds(robot, configurations)
+
+        # removes the `None` ones
+        if options and "cull" in options and options["cull"] is True:
+            configurations = [c for c in configurations if c is not None]
+
+        return configurations
+
+    def _inverse_kinematics(self, frame):
+        pass
+
+    def joint_angles_to_configurations(self, robot, solutions):
+        joint_names = robot.get_configurable_joint_names()
+        return [Configuration.from_revolute_values(q, joint_names=joint_names) for q in solutions]
+
+    def try_to_fit_configurations_between_bounds(self, robot, configurations):
+        """
+        """
+        j1, j2, j3, j4, j5, j6 = robot.get_configurable_joints()
+        for i, c in enumerate(configurations):
+            if c is None:
+                continue
+            a1, a2, a3, a4, a5, a6 = c.values()
+            try:
+                a1 = fit_within_bounds(a1, j1.limit.lower, j1.limit.upper)
+                a2 = fit_within_bounds(a2, j2.limit.lower, j2.limit.upper)
+                a3 = fit_within_bounds(a3, j3.limit.lower, j3.limit.upper)
+                a4 = fit_within_bounds(a4, j4.limit.lower, j4.limit.upper)
+                a5 = fit_within_bounds(a5, j5.limit.lower, j5.limit.upper)
+                a6 = fit_within_bounds(a6, j6.limit.lower, j6.limit.upper)
+                configurations[i].joint_values = [a1, a2, a3, a4, a5, a6]
+            except AssertionError:
+                configurations[i] = None
+        return configurations
+
+
+class UR5_Analytical_IK(Analytical_InverseKinematics):
+
+    def _inverse_kinematics(self, frame):
+        from compas_fab.backends.kinematics.offset_wrist_kinematics import UR5
+        return UR5().inverse(frame)
+
+
+if __name__ == "__main__":
+    import time
+    import compas_fab
+    from compas_fab.robots import Tool
+    from compas.datastructures import Mesh
+    from compas.geometry import Point, Vector
+    from compas_fab.robots import RobotSemantics
+    from compas_fab.backends import PyBulletClient
+
+    class Client(PyBulletClient):
+        def inverse_kinematics(self, *args, **kwargs):
+            return UR5_Analytical_IK(self)(*args, **kwargs)
```

Comment:
In regards to the link, this clearly is one way but it would be better if there would be a way to automatically map the robot model to one of these ik solvers.